src/libostree/ostree-repo-commit.c

```diff
@@ -3286,22 +3286,32 @@ _ostree_repo_commit_modifier_apply (OstreeRepo               *self,
                                     GFileInfo                *file_info,
                                     GFileInfo               **out_modified_info)
 {
+  gboolean canonicalize_perms = FALSE;
+  gboolean has_filter = FALSE;
   OstreeRepoCommitFilterResult result = OSTREE_REPO_COMMIT_FILTER_ALLOW;
   GFileInfo *modified_info;
 
-  if (modifier == NULL ||
-      (modifier->filter == NULL &&
-       (modifier->flags & OSTREE_REPO_COMMIT_MODIFIER_FLAGS_CANONICAL_PERMISSIONS) == 0))
+  /* Auto-detect bare-user-only repo, force canonical permissions. */
+  canonicalize_perms |= (self->mode == OSTREE_REPO_MODE_BARE_USER_ONLY);
```

Comment:
Minor: this is OK, though to be consistent with the rest of the codebase might be worth avoiding bitwise ops for these pseudo-booleans.

Reply:
How would you prefer to write this? `if (mode is user-only) { canonicalize_perms = TRUE }`?

Reply:
Yup, exactly! With a newline and less braces. :)